driver-bitfury16.c

```diff
@@ -0,0 +1,4589 @@
+	bf_cmd_buffer_t* cmd_buffer = &info->chipboard[board_id].cmd_buffer;
+
+	if (cmd_buffer->status == EMPTY) {
+		if ((opt_bf16_renonce == RENONCE_DISABLED) ||
+			(renonce_chip_address[board_id].board_id == -1)) {
+
+			fill_cmd_buffer_loop(bitfury, board_id, false, 0);
+		} else {
+			L_LOCK(info->renonce_list);
+			bf_data_t* rdata = info->renonce_list->head;
+			while (rdata != NULL) {
+				if (RENONCE(rdata)->sent == false)
+					renonce_count++;
+				rdata = rdata->next;
+			}
+			L_UNLOCK(info->renonce_list);
+
+			if (do_renonce[board_id] < RENONCE_SEND) {
+				fill_cmd_buffer_loop(bitfury, board_id, true, renonce_count);
+
+				do_renonce[board_id]++;
+			} else {
+				if (renonce_count >= RENONCE_COUNT) {
+					renonce_task_update(bitfury, board_id, RENONCE_COUNT);
+
+					do_renonce[board_id] = 0;
+					cmd_buffer->status = TX_READY;
+				} else {
+					fill_cmd_buffer_loop(bitfury, board_id, true, renonce_count);
+
+					do_renonce[board_id]++;
+				}
+			}
+		}
+	}
+}
+
+static uint8_t update_chip_status(struct cgpu_info *bitfury, bf_cmd_status_t cmd_status)
+{
+	struct bitfury_info *info = (struct bitfury_info *)(bitfury->device_data);
+	uint8_t board_id  = cmd_status.chip_address.board_id;
+	uint8_t bcm250_id = cmd_status.chip_address.bcm250_id;
+	uint8_t chip_id   = cmd_status.chip_address.chip_id;
+
+	switch (info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].status) {
+
+	case UNINITIALIZED:
+	case TOGGLE_SET:
+	case CLOCK_SET:
+	case MASK_SET:
+		applog(LOG_DEBUG, "%s: chipworker_thr: chip [%d:%d:%2d], send_cmd [%s]",
+				bitfury->drv->name,
+				board_id, bcm250_id, chip_id,
+				get_cmd_description(cmd_status.cmd_code));
+
+		/* update chip status */
+		if (cmd_status.checksum_error == false)
+			info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].status++;
+
+		else {
+#ifndef DISABLE_SEND_CMD_ERROR
+			applog(LOG_ERR, "%s: chipworker_thr: chip [%d:%d:%2d]: error send_cmd [%s]. "
+					"checksum: expected: [%02x]; received: [%02x]",
+					bitfury->drv->name,
+					board_id, bcm250_id, chip_id,
+					get_cmd_description(cmd_status.cmd_code),
+					cmd_status.checksum_expected,
+					cmd_status.checksum_received);
+#endif
+
+			/* increase error counters */
+			increase_errors(info, cmd_status.chip_address);
+		}
+		break;
+
+	case TASK_SENT:
+		applog(LOG_DEBUG, "%s: chipworker_thr: chip [%d:%d:%2d], send_cmd [%s]",
+				bitfury->drv->name,
+				board_id, bcm250_id, chip_id,
+			 	get_cmd_description(cmd_status.cmd_code));
+
+		/* read task status from chip*/
+		if (cmd_status.checksum_error == false) {
+			uint8_t new_buff = ((cmd_status.status & 0x0f) == 0x0f) ? 1 : 0;
+
+			/* status cmd counter */
+			info->chipboard[board_id].bcm250[bcm250_id].chips[chip_id].status_cmd_dx++;
+			info->chipboard[board_id].bcm250[bcm250_id].status_cmd_dx++;
+			info->chipboard[board_id].status_cmd_dx++;
+			info->status_cmd_dx++;
+
+			/* check if chip task has switched */
+			if (new_buff != info->chipboard[board_id].bc
```

Comment:
driver-bitfury16.c:3218:23: warning: comparison is always true due to limited range of data type [-Wtype-limits]
        if ((bcm250_id >= 0) && (bcm250_id < BCM250_NUM / 2)) {